lib/auth/clt.go

```diff
@@ -1176,17 +1176,29 @@ func (c *Client) CreateSAMLAuthRequest(req services.SAMLAuthRequest) (*services.
 }
 
 // ValidateSAMLResponse validates response returned by SAML identity provider
-func (c *Client) ValidateSAMLResponse(re string) (*SAMLAuthResponse, error) {
+func (c *Client) ValidateSAMLResponse(re string) (*SAMLAuthResponse, *SsoDiagnosticInfo, error) {
 	out, err := c.PostJSON(c.Endpoint("saml", "requests", "validate"), validateSAMLResponseReq{
 		Response: re,
 	})
+
+	// try to recover diagnostic info from proxy error
+	var di *SsoDiagnosticInfo
+	di = nil
+
+	var ssoDi SsoDiagnosticInfo
+	if trace.UnwrapProxyField(err, "sso-diag-info", &ssoDi) {
```

Comment:
missing error check.

Reply:
UnwrapProxyField returns boolean, not an error?